src/libstd/sys_common/remutex.rs

```diff
@@ -70,7 +75,7 @@ impl<T> ReentrantMutex<T> {
     /// If another user of this mutex panicked while holding the mutex, then
     /// this call will return failure if the mutex would otherwise be
     /// acquired.
-    pub fn lock(&self) -> LockResult<ReentrantMutexGuard<'_, T>> {
+    pub fn lock(&self) -> ReentrantMutexGuard<'_, T> {
```

Comment:
I removed the poisioning here since this type is only used for one thing in libstd, I/O handles, which ignore poisoning anyway.